lib/application.js

```diff
@@ -219,11 +219,19 @@ app.models = function() {
  * @param {Object} config The data source config
  */
 app.dataSource = function(name, config) {
-  var ds = dataSourcesFromConfig(name, config, this.connectors, this.registry);
-  this.dataSources[name] =
-  this.dataSources[classify(name)] =
-  this.dataSources[camelize(name)] = ds;
-  return ds;
+  try {
+    var ds = dataSourcesFromConfig(name, config, this.connectors, this.registry);
+    this.dataSources[name] =
+    this.dataSources[classify(name)] =
```

Comment:
> I regret adding this code... :(

I think this particular line is safe to be removed in 3.0 - feel free to send a PR against loopback's master. Just add a short explanation to 3.0-RELEASE-NOTES.md as part of that patch.